quiz-microsevice/data-access/repositories/client-repository.js

```diff
@@ -12,8 +14,34 @@ const createOne = async (name, clientId, clientSecretHash) => {
 
 const retrieveOneByClientId = async (clientId) => {
   const client = await Client.findOne({ clientId });
-
+  
   return client ? toDTO(client) : null;
 };
 
-module.exports = { createOne, retrieveOneByClientId };
+const retrieveAllClients = async () => {
+  
+  const clients = await Client.find().select('-clientSecretHash');
+  
+  return clients.length > 0 ? clients : [];
+  
+  
+};
+
+/* 
+Here I make GET request to get ObjectID, not clientId.
+So I'm using (findById) method, not findOne.
+*/
+const regenerateClientCredentials = async(id, clientId, clientSecretHash) => {
+  const client = await Client.findById(id);
+  if(!client){
+    // console.log('hello')
+    throw new NotExistError("There is no client with this ObjectID.");
```

Comment:
Don't throw business-logic errors within data-access layer just return null, Then check the value in business logic layer if its null throw NotExsitError.

Reply:
![Screenshot (768)](https://github.com/user-attachments/assets/faa50b79-17a7-413d-899d-b08b14cf438a)
So do you need me to make validation in this function by id parameter?